System/Console/Haskeline/Backend/Win32/MinTTY.hsc

```diff
@@ -0,0 +1,235 @@
+{-
```

Comment:
Given that the module already exists in newer versions of Win32, I'd prefer to do something simpler:
- Only do this fix when Win32>=2.5
- Add a comment to getInputPassword describing the bug in the meantime

Otherwise, I'm concerned about the extra level of complexity we're adding in order to fix this corner case (and my ability to maintain it in the long run), especially since that there's already other ways Haskeline behaves suboptimally on MinTTY.